File: protoif/audio.js
// audio screen handling

var audio = {
  handle_msg: function(msg) {
    if (msg.Audio) {
      $("#audio-not-connected").toggle(!msg.Audio.connected);
      $("#audio-player").toggle(msg.Audio.connected);
      
      $("#audio-song-title").text(msg.Audio.song.title);
      $("#audio-song-artist").text(msg.Audio.song.artist);
    }
  }
};

registerModule(audio);

$("#audio-playpause-btn").click(function() {
  // send play/pause to server
  console.log("playpause");
  sendMsg({
    "audio": {
      "cmd": "playpause"
    }
  });
});

$("#audio-next-btn").click(function() {
  // send next to server
  console.log("next");
  sendMsg({
    "audio": {
      "cmd": "next"
    }
  });
});

$("#audio-prev-btn").click(function() {
  // send prev to server
  console.log("prev");
  sendMsg({
    "audio": {
      "cmd": "prev"
    }
  });
});

File: protoserver/audio.py
import json


class Audio:
  def get_state(self):
    state = {
      "connected": True,
      "paused": False,
      "song": {
        "title": "title",
        "artist": "artist"
      }
    }
    return state

  def handle_msg(self, msg):
    cmd = msg["audio"]["cmd"]
    if cmd == "playpause":
      print "playpause"
    elif cmd == "next":
      print "next track"
    elif cmd == "prev":
      print "prev track"
